### PR TITLE
sdosapat/vz 6338: Collect additional resources for bug-report CLI from all namespaces where component failed

### PR DESCRIPTION
**Pull request by saidu941**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

Summary:
> This PR have the changes to create a config map for componentNames to Namespaces in global level constants
> This will add logic to find the failed components, and then the corresponding namespaces list, and use that namespaces list to collect additional resources for the bug-report CLI command

We will take the changes for VPO _Component.go to use constants from pkg/constants.go in the follow up PR

### CODE REVIEW

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -3,7 +3,10 @@
 
 package constants
 
-import "time"
+import (
+	PlatformOperatorConstants "github.com/verrazzano/verrazzano/platform-operator/constants"
```
Comment (kminder) [inline pkg/constants/constants.go:7]: Most other package aliases are short lowercase abbreviations.

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -137,3 +140,64 @@ const MetricsTemplateAPIVersion = "app.verrazzano.io/v1alpha1"
...
+	weblogic_operator:               {VerrazzanoSystemNamespace},
+	verrazzano_authproxy:            {VerrazzanoSystemNamespace},
+	istio:                           {IstioSystemNamespace},
+	external_dns:                    {CertManagerNamespace},
+	verrazzano_application_operator: {VerrazzanoSystemNamespace},
+	coherence_operator:              {VerrazzanoSystemNamespace},
+	ingress_controller:              {PlatformOperatorConstants.IngressNginxNamespace},
+	mysql:                           {KeycloakNamespace},
+	cert_manager:                    {CertManagerNamespace},
+	rancher:                         {RancherSystemNamespace}, // TODO vz-6833 add multiple namespaces
```
Comment (kminder) [inline pkg/constants/constants.go:187]: This ticket doesn't exist.

**Comment by kminder** [comment]
Please provide a better PR description.

File: tools/vz/pkg/bugreport/genreport.go [hunk 3]
```diff
@@ -75,8 +77,15 @@ func captureVerrazzanoResources(client clipkg.Client, kubeClient kubernetes.Inte
 		return err
 	}
 
+	//TODO vz-6338 1. GetProblemPods 2. GetNamespacesListFromProblemPods //Note to remove dup
```
Comment (kminder) [inline tools/vz/pkg/bugreport/genreport.go:80]: What does this TODO mean?  Should it be in the code at all?

File: tools/vz/pkg/helpers/vzhelper.go [hunk 4]
```diff
@@ -100,3 +101,54 @@ func NewScheme() *runtime.Scheme {
 	_ = appv1.SchemeBuilder.AddToScheme(scheme)
 	return scheme
 }
+
+func GetAllUniqueNameSpacesForFailedComponents(client client.Client) ([]string, error) {
```
Comment (kminder) [inline tools/vz/pkg/helpers/vzhelper.go:105]: Needs a function comment.  This function name implies that only namespaces for components that have "failed" will be returned.  Consider renaming.

File: tools/vz/pkg/helpers/vzhelper.go [hunk 5]
```diff
@@ -100,3 +101,54 @@ func NewScheme() *runtime.Scheme {
...
+			if !nsListMap[eachNameSpace] {
+				nsList = append(nsList, eachNameSpace)
+				nsListMap[eachNameSpace] = true
+			}
+		}
+	}
+	return nsList, err
+}
+
+// Read the Verrazzano resource and return the list of components which did not reach Ready state
```
Comment (kminder) [inline tools/vz/pkg/helpers/vzhelper.go:125]: Make function comment godoc compliant.

File: tools/vz/pkg/helpers/vzhelper.go [hunk 6]
```diff
@@ -100,3 +101,54 @@ func NewScheme() *runtime.Scheme {
...
+					continue
+				}
+				compsNotReady = append(compsNotReady, compStatusDetail.Name)
+			}
+		}
+	}
+	return compsNotReady, nil
+}
+
+func getNameSpacesByComponent(componentName string) []string {
```
Comment (kminder) [inline tools/vz/pkg/helpers/vzhelper.go:148]: I don't think this function has any value.  Go lang returns a default zero value of nil for slices.  So these are equivalent I think.
value, _ := constants.ComponentNameToNamespacesMap[componentName]
value := getNameSpacesByComponent(componentName)

File: pkg/constants/constants.go [hunk 7]
```diff
@@ -137,3 +140,64 @@ const MetricsTemplateAPIVersion = "app.verrazzano.io/v1alpha1"
 
 // SecretKind is the kind for a secret
 const SecretKind = "Secret"
+
+// Components Names
+const (
```
[the remark this answers is not in the document]
Comment (saidu941) [reply pkg/constants/constants.go:145]: Updated the PR with changes

File: pkg/constants/constants.go (hunk 1 above)
Comment (saidu941) [reply pkg/constants/constants.go]: Updated the PR with correction

File: tools/vz/pkg/bugreport/genreport.go (hunk 3 above)
Comment (saidu941) [reply tools/vz/pkg/bugreport/genreport.go]: Cleanedup the code and updated PR

File: tools/vz/pkg/helpers/vzhelper.go (hunk 5 above)
Comment (saidu941) [reply tools/vz/pkg/helpers/vzhelper.go]: Updated the PR

File: tools/vz/pkg/helpers/vzhelper.go (hunk 6 above)
Comment (saidu941) [reply tools/vz/pkg/helpers/vzhelper.go]: Updated the PR

File: tools/vz/pkg/helpers/vzhelper.go (hunk 4 above)
Comment (agv426) [reply tools/vz/pkg/helpers/vzhelper.go]: The function is intended to give a list of all namespaces for components which are "Not Ready" (ie: they may not be failed, they may simply be in progress. So maybe name it something like GetNamespacesForNotReadyComponents (removed the unique, and all to trim the name down a bit as well)

**Review by agv426 — DISMISSED** [review]

**Review by bhatpmk — APPROVED** [review]

**Pull request merged**